qiskit_nature/second_q/hamiltonians/quadratic_hamiltonian.py

```diff
@@ -123,29 +123,40 @@ def __init__(
                 if not _is_antisymmetric(antisymmetric_part, rtol=rtol, atol=atol):
```

Comment:
I cannot comment higher up, so I am leaving this here: we started to constrain some arguments to be keyword-only in our API by inserting the `*` catch all positional arguments in the argument list.
I think, this would also be useful here, seeing things like `validate`, `atol`, `rtol`...
Personally, I would make it the following:
```python
    def __init__(
        self,
        hermitian_part: np.ndarray | None = None,
        antisymmetric_part: np.ndarray | None = None,
        *,
        constant: float = 0.0,
        num_modes: int | None = None,
        validate: bool = True,
        rtol: float = 1e-5,
        atol: float = 1e-8,
    ) -> None:
```
But I would be open to having `constant` and/or `num_modes` above the `*` catchall.

Reply:
Some more observation with respect to types:
- constant is constrained to float and does not allow complex. Is this intentional?
- you constrain the matrices to be dense. It might be useful to open this up to `sparse.SparseArray`, too (this would have more implications down below, which I will not comment on any more than stating it here)

Reply:
I added the asterisk to force keyword arguments.

>constant is constrained to float and does not allow complex. Is this intentional?

Yes. The constant must be real because the Hamiltonian is hermitian.

>you constrain the matrices to be dense. It might be useful to open this up to sparse.SparseArray, too (this would have more implications down below, which I will not comment on any more than stating it here)

Let's leave that for the future.